src/onnx/parse_split.cpp

```diff
@@ -68,13 +68,25 @@ struct parse_split : op_parser<parse_split>
         // no split attribute, input is equally divided
         else
         {
-            if((lens[tuned_axis] % info.num_outputs) != 0)
+            std::size_t num_outputs = info.num_outputs;
+            // the num_outputs attribute seems to be redundant since we already have
+            // node_info::num_outputs, but we can still perform an error check
+            if(contains(info.attributes, "num_outputs"))
             {
-                MIGRAPHX_THROW("PARSE_SPLIT: input cannot be equally divided into " +
-                               std::to_string(info.num_outputs) + " splits!");
+                num_outputs =
+                    parser.parse_value(info.attributes.at("num_outputs")).at<std::size_t>();
+                if(num_outputs != info.num_outputs)
+                {
+                    MIGRAPHX_THROW("PARSE_SPLIT: num_outputs attribute " +
+                                   std::to_string(num_outputs) +
+                                   " doesn't match actual number of outputs " +
+                                   std::to_string(info.num_outputs) + "!");
+                }
             }
-            auto dl = lens[tuned_axis] / info.num_outputs;
-            vec_splits.resize(info.num_outputs, dl);
+
+            auto dl = lens[tuned_axis] / num_outputs;
+            vec_splits.resize(lens[tuned_axis] % num_outputs, dl + 1);
+            vec_splits.resize(num_outputs, dl);
```

Comment:
It's not clear to me if this is the intended behavior from ONNX. The specification is not clear about how to handle uneven splits and only says the last "chunk" will be smaller. Numpy's `split` function raises an error when it cannot be split evenly.

Reply:
I agree that the ONNX spec is not clear in what they mean by "chunk". I based this solution on Numpy's [`array_split`](https://numpy.org/doc/stable/reference/generated/numpy.array_split.html).